app/models/assignment.rb

```diff
@@ -57,32 +63,34 @@ class Assignment < Assessment
   accepts_nested_attributes_for :assignment_files, allow_destroy: true
   validates_associated :assignment_files
 
-  # this has to be before :peer_reviews or it throws a HasManyThroughOrderError
-  has_many :groupings, foreign_key: :assessment_id, inverse_of: :assignment
   # Assignments can now refer to themselves, where this is null if there
   # is no parent (the same holds for the child peer reviews)
   belongs_to :parent_assignment,
              class_name: 'Assignment', optional: true, inverse_of: :pr_assignment, foreign_key: :parent_assessment_id
-  has_one :pr_assignment, class_name: 'Assignment', foreign_key: :parent_assessment_id, inverse_of: :parent_assignment
+  has_one :pr_assignment,
+          class_name: 'Assignment',
+          dependent: :destroy,
+          foreign_key: :parent_assessment_id,
+          inverse_of: :parent_assignment
   has_many :peer_reviews, through: :groupings
   has_many :pr_peer_reviews, through: :parent_assignment, source: :peer_reviews
 
-  has_many :current_submissions_used, through: :groupings,
-                                      source: :current_submission_used
+  has_many :current_submissions_used,
```

Comment:
you can revert the changes here (they are now purely stylistic since you removed the `dependent: :destroy`